engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformation/TransformContext.java

```diff
@@ -24,6 +25,7 @@ public final class TransformContext {
   private final Map<DirectBuffer, ExecutableProcess> processes = new HashMap<>();
   private final Map<DirectBuffer, ExecutableMessage> messages = new HashMap<>();
   private final Map<DirectBuffer, ExecutableError> errors = new HashMap<>();
+  private final Map<DirectBuffer, ExecutableCatchEventElement> linkCatchEvents = new HashMap<>();
```

Comment:
:wrench: We could align with the existing implementation by storing `ExecutableLink` instead of `ExecutableCatchEventElement`.

And returning the `ExecutableLink` on calling `getLink()`.